test/integration/consul-container/test/resource/http_api/acl_enabled_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resource
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_ServerAgent_Endpoints(t *testing.T) {
+	t.Parallel()
+
+	numOfServers := 1
+	numOfClients := 0
+	cluster, resourceClient := SetupClusterAndClient(t, makeClusterConfig(numOfServers, numOfClients, true), true)
+
+	resource := Resource{
+		HttpClient: resourceClient,
+	}
+
+	defer Terminate(t, cluster)
+
+	testCases := []testCase{
+		{
+			description: "should write resource successfully when token is provided",
+			operations: []operation{
+				{
+					action:           applyResource,
+					expectedErrorMsg: "",
+					includeToken:     true,
+				},
+				{
+					action:           readResource,
+					expectedErrorMsg: "",
+					includeToken:     true,
+				},
+				{
+					action:           listResource,
+					expectedErrorMsg: "",
+					includeToken:     true,
+				},
+				{
+					action:           deleteResource,
+					expectedErrorMsg: "",
+					includeToken:     true,
+				},
+			},
+			config: []config{
+				{
+					gvk:          demoGVK,
+					resourceName: "korn",
+					queryOptions: defaultTenancyQueryOptions,
+					payload:      demoPayload,
+				},
+				{
+					gvk:          demoGVK,
+					resourceName: "korn",
+					queryOptions: defaultTenancyQueryOptions,
+				},
+				{
+					gvk:          demoGVK,
+					queryOptions: defaultTenancyQueryOptions,
+				},
+				{
+					gvk:          demoGVK,
+					resourceName: "korn",
+					queryOptions: defaultTenancyQueryOptions,
+				},
+			},
+		},
+		{
+			description: "should return permission denied",
+			operations: []operation{
+				{
+					action:           applyResource,
+					expectedErrorMsg: "Unexpected response code: 403 (rpc error: code = PermissionDenied desc = Permission denied",
+					includeToken:     false,
+				},
+				{
+					action:           applyResource,
+					expectedErrorMsg: "",
+					includeToken:     true,
+				},
+				{
+					action:           readResource,
+					expectedErrorMsg: "Unexpected response code: 403 (rpc error: code = PermissionDenied desc = Permission denied",
+					includeToken:     false,
+				},
+				{
+					action:           listResource,
+					expectedErrorMsg: "Unexpected response code: 403 (rpc error: code = PermissionDenied desc = Permission denied",
+					includeToken:     false,
+				},
+				{
+					action:           deleteResource,
+					expectedErrorMsg: "Unexpected response code: 403 (rpc error: code = PermissionDenied desc = Permission denied",
+					includeToken:     false,
+				},
+			},
+			config: []config{
+				{
+					gvk:          demoGVK,
+					resourceName: "prince",
+					queryOptions: defaultTenancyQueryOptions,
+					payload:      demoPayload,
+				},
```

Comment:
I'm wondering if we can merge this with the individual operations above? As each test case uses the same gvk/resourceName/queryOptions - Maybe they can just be hardcoded when we loop through `tc.operations`?

Reply:
Hmm... we cannot hard code the `resourceName` since that would change depending on the operation. Also, for your tenancy cases I am guessing you will be passing your own tenancy values.

Reply:
Ah yes, my mistake!